### ISSUE TITLE
Why was no security advisory issued for SQL injection fixed in 4.4.3

### ISSUE THREAD

**Issue by iamjoeker** [nestjsx/crud#599]
The change notes for [4.4.3](https://github.com/nestjsx/crud/releases/tag/v4.4.3) indicates that a critical bug exposing multiple vectors for SQL injection was fixed as part of the release. My team discovered this just by chance this evening as we started to add a new feature to a module that uses the CrudController. 

This brought up an important question: why was there no security advisory issued when this critical bug was identified? Is there a process in place for this project to issue advisories when security issues are reported?

**Comment by zMotivat0r** [comment]
>why was there no security advisory issued when this critical bug was identified?

no security advisory was issued because the fix contained the lib's internal methods updates without any need for library users to change their code or the way they use this library.

>Is there a process in place for this project to issue advisories when security issues are reported?

this particular case is the only one that deals with security and I've fixed it as soon as I got a report from the third party security specialist. If they're gonna be another security issue I'll do the same - will fix it asap and inform library users and issue advisories if there are any needs.

**Comment by dschoeni** [comment]
> no security advisory was issued because the fix contained the lib's internal methods updates without any need for library users to change their code or the way they use this library.

It would be great if you could issue notices or warnings for such bugs over appropriate channels. Many are using this library in production, and it is of utmost importance that people update their library to get the fix in, especially with SQL Injection bugs.

I know it may look bad from the outside if a library urges its users to update, but its important especially in those cases.

**Comment by chriszrc** [comment]
@zMotivat0r I agree with @dschoeni , I expected to be notified about this at least when I ran npm audit. It looks like the way to get an npm audit submitted is through the "Report Malware" button the npm package page, which goes here:

https://www.npmjs.com/support?inquire=security&security-inquire=malware&package=%40nestjsx%2Fcrud&version=4.4.3

Maybe you could do this as the author?
